### PR TITLE
Enable ONNX test in CI

### PR DESCRIPTION
**Pull request by msaroufim**
ONNX is one of our most popular model optimization runtimes, we shouldn't disable this test in CI by default anymore, it's hiding basic failures

Our CI takes 40min and our regression tests take 1h and for context the PyTorch CI takes around 4h. Would just suggest we run regression tests on CI all the time otherwise we get dumb bugs

Fixes https://github.com/pytorch/serve/issues/2360

### CODE REVIEW

File: requirements/developer.txt [hunk 1]
```diff
@@ -16,3 +16,6 @@ twine
 mypy
 torchpippy
 intel_extension_for_pytorch; sys_platform != 'win32' and sys_platform != 'darwin'
+onnx-runtime
+psutil
```
Comment (msaroufim) [inline requirements/developer.txt:20]: adding psutil and pyyaml here since otherwise they're not being installed, not sure why yet

**Review by agunapal — APPROVED** [review]

**Pull request merged**